Tests/Runtime/MobiledgeX_RuntimeTests.cs

```diff
@@ -29,107 +39,107 @@ public void MobiledgeXEnvironmentSetup()
+            mix.appName = appName;
+            mix.appVers = appVers;
+            mix.orgName = orgName;
             var task = Task.Run(async () =>
             {
-                return await RegisterHelper();
+                return await RegisterHelper(mix);
             });
 
             Debug.Log("result of registerClient is " + task.Result);
             Assert.True(task.Result);
         }
 
         [Test]
-        [TestCase("MobiledgeX", "MobiledgeX SDK Demo", "2.0")]
+        [TestCase("MobiledgeX-Samples", "sdktest", "9.0")]
         public void FindCloudlet(string orgName, string appName, string appVers)
         {
-            integration.orgName = orgName;
-            integration.appName = appName;
-            integration.appVers = appVers;
+            MobiledgeXIntegration mix = new MobiledgeXIntegration(new CarrierInfoClass(), null, new UniqueIDClass(), new TestDeviceInfo());
+            mix.appName = appName;
+            mix.appVers = appVers;
+            mix.orgName = orgName;
             var task = Task.Run(async () =>
             {
-                bool registered = await RegisterHelper();
+                bool registered = await RegisterHelper(mix);
                 Assert.True(registered, "Unable to register");
-                return await FindCloudletHelper();
+                return await FindCloudletHelper(mix);
             });
 
             Debug.Log("result of findCloudlet is " + task.Result);
             Assert.True(task.Result);
         }
 
         [Test]
-        [TestCase("MobiledgeX", "MobiledgeX SDK Demo", "2.0")]
-        public void GetRestUrl(string orgName, string appName, string appVers)
+        [TestCase("MobiledgeX-Samples", "sdktest", "9.0", "http", 8085)]
+        [TestCase("MobiledgeX-Samples", "sdktest", "9.0", "https", 2015)]
+        [TestCase("MobiledgeX-Samples", "sdktest", "9.0", "tcp", 2016)]
+        [TestCase("MobiledgeX-Samples", "sdktest", "9.0", "ws", 3765)]
+        [TestCase("MobiledgeX-Samples", "sdktest", "9.0", "udp", 2015)]
+        public void GetUrl(string orgName, string appName, string appVers, string proto, int port)
         {
-            integration.orgName = orgName;
-            integration.appName = appName;
-            integration.appVers = appVers;
+            MobiledgeXIntegration mix = new MobiledgeXIntegration(new CarrierInfoClass(), null, new UniqueIDClass(), new TestDeviceInfo());
+            mix.appName = appName;
+            mix.appVers = appVers;
+            mix.orgName = orgName;
             var task = Task.Run(async () =>
             {
-                return await GetUrlHelper(orgName, appName, appVers, "http");
+                return await GetUrlHelper(mix, proto, port);
             });
-
-            Assert.AreEqual(task.Result, "http://mobiledgexsdkdemo-tcp.mobiledgexmobiledgexsdkdemo20.sdkdemo-app-cluster.us-los-angeles.gcp.mobiledgex.net:8008"); 
+            Debug.Log(task.Result);
+            Assert.IsNotEmpty(task.Result);
```

Comment:
Can we check for the contents of the string? eg. if proto is "ws", check that the url has "ws://" and the port is there. Maybe you can have a switch statement for each of the proto cases. Same for the GetHost test

Reply:
Fixed I checked for proto and port in the url created